src/script/components/MessagesList/Message/ContentMessage/asset/TextMessageRenderer.tsx

```diff
@@ -0,0 +1,186 @@
+/*
+ * Wire
+ * Copyright (C) 2021 Wire Swiss GmbH
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program. If not, see http://www.gnu.org/licenses/.
+ *
+ */
+
+import {useEffect, useRef, FC} from 'react';
+
+import {Text} from 'src/script/entity/message/Text';
+import {handleKeyDown} from 'Util/KeyboardUtil';
+import {useDisposableRef} from 'Util/useDisposableRef';
+
+export type ElementType = 'markdownLink' | 'email' | 'mention';
+
+interface TextMessageRendererProps {
+  onClickMsg: (event: MouseEvent | KeyboardEvent, asset: Text, elementType: ElementType) => void;
+  text: string;
+  isCurrentConversationFocused: boolean;
+  msgClass: string;
+  asset: Text;
+  isQuoteMsg?: boolean;
+  edited_timestamp?: number;
+  setCanShowMore?: (showMore: boolean) => void;
+}
+const events = ['click', 'keydown', 'auxclick'];
+
+export const TextMessageRenderer: FC<TextMessageRendererProps> = ({
+  text,
+  onClickMsg,
+  msgClass,
+  isCurrentConversationFocused,
+  asset,
+  isQuoteMsg = false,
+  edited_timestamp,
+  setCanShowMore,
+  ...props
+}) => {
+  const containerRef = useRef<HTMLDivElement | null>(null);
+  const detectLongQuotes = useDisposableRef(
+    element => {
+      const preNode = element.querySelector('pre');
+      const width = Math.max(element.scrollWidth, preNode ? preNode.scrollWidth : 0);
+      const height = Math.max(element.scrollHeight, preNode ? preNode.scrollHeight : 0);
+      const isWider = width > element.clientWidth;
+      const isHigher = height > element.clientHeight;
+      setCanShowMore?.(isWider || isHigher);
+      return () => {};
+    },
+    [edited_timestamp],
+  );
+
+  useEffect(() => {
+    if (!containerRef.current) {
+      return undefined;
+    }
+    const emailLinks = containerRef.current && [...containerRef.current.querySelectorAll('[data-email-link]')];
+    const linkTarget = containerRef.current && [...containerRef.current.querySelectorAll('a')];
```

Comment:
Proper name for this should be linkTargets
```suggestion
    const linkTargets = containerRef.current && [...containerRef.current.querySelectorAll('a')];
```